modules/service/src/main/scala/lucuma/odb/logic/TimeEstimateService2.scala

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2016-2025 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.logic
+
+import cats.Order
+import cats.Order.catsKernelOrderingForOrder
+import cats.Semigroup
+import cats.data.OptionT
+import cats.effect.Concurrent
+import cats.kernel.CommutativeMonoid
+import cats.syntax.apply.*
+import cats.syntax.foldable.*
+import cats.syntax.functor.*
+import cats.syntax.functorFilter.*
+import cats.syntax.monoid.*
+import eu.timepit.refined.types.numeric.NonNegShort
+import lucuma.core.enums.ScienceBand
+import lucuma.core.model.Group
+import lucuma.core.model.Observation
+import lucuma.core.model.Program
+import lucuma.core.model.sequence.CategorizedTime
+import lucuma.core.model.sequence.CategorizedTimeRange
+import lucuma.core.util.CalculatedValue
+import lucuma.core.util.CalculationState
+import lucuma.itc.client.ItcClient
+import lucuma.odb.data.GroupTree
+import lucuma.odb.sequence.util.CommitHash
+import lucuma.odb.service.ObscalcService
+import lucuma.odb.service.Services
+import lucuma.odb.service.Services.Syntax.*
+import skunk.Transaction
+
+/**
+ * A service that can estimate the time required to finish the unexecuted
+ * remainder of a program.
+ */
+sealed trait TimeEstimateService2[F[_]]:
+
+  /**
+   * Estimates the remaining time for all observations in the program that are
+   * well defined.
+   */
+  def estimateProgramRange(
+    programId: Program.Id
+  )(using Transaction[F]): F[Option[CalculatedValue[CategorizedTimeRange]]]
+
+  /**
+   * Estimates the remaining time for all observations in the group that are
+   * well defined.
+   */
+  def estimateGroupRange(
+    groupId: Group.Id
+  )(using Transaction[F]): F[Option[CalculatedValue[CategorizedTimeRange]]]
+
+  /**
+   * Estimates the remaining time for all observations in the program by band,
+   * ignoring `minRequired` in groups.
+   */
+  def estimateProgramBanded(
+    programId: Program.Id
+  )(using Transaction[F]): F[Map[Option[ScienceBand], CalculatedValue[CategorizedTime]]]
+
+  /**
+   * Estimates the remaining time for all observations in the group by band,
+   * ignoring `minRequired`.
+   */
+  def estimateGroupBanded(
+    groupId: Group.Id
+  )(using Transaction[F]): F[Map[Option[ScienceBand], CalculatedValue[CategorizedTime]]]
+
+
+object TimeEstimateService2:
+
+  def instantiate[F[_]: Concurrent](
+    commitHash:   CommitHash,
+    itcClient:    ItcClient[F],
+    calculator:   TimeEstimateCalculatorImplementation.ForInstrumentMode,
+  )(using Services[F]): TimeEstimateService2[F] =
+    lazy val obscalcService: ObscalcService[F] =
+      services.obscalcService(commitHash, itcClient, calculator)
+
+    new TimeEstimateService2[F]:
+
+      // OBSCALC TODO: In core, need to fiddle with `NotGiven` in `CalculatedValue`
+      // to avoid the need to do this.
+      given Semigroup[CalculatedValue[CategorizedTime]] =
```

Comment:
You can fix this by slicing the givens into interfaces of increasing generality. This will provide the most specific instance you ask for.

```scala
case class Foo[A](a: A)

object Foo extends FooSlice1:
    given [A: CommutativeMonoid]: CommutativeMonoid[Foo[A]] = ???

trait FooSlice1 extends FooSlice2:
  given [A: Monoid]: Monoid[Foo[A]] = ???

trait FooSlice2:
  given [A: Semigroup]: Semigroup[Foo[A]] = ???
```